File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/dto/entity/lang/response/LangResponseDTO.java
package ru.herooo.mylanguageweb.dto.entity.lang.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import ru.herooo.mylanguageweb.controllers.json.LongDeserializer;
import ru.herooo.mylanguageweb.controllers.json.LongSerializer;
import ru.herooo.mylanguageweb.dto.entity.country.response.CountryResponseDTO;

public class LangResponseDTO {
    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("id")
    private long id;

    @JsonProperty("title")
    private String title;

    @JsonProperty("code")
    private String code;

    @JsonProperty("is_active_for_in")
    private boolean isActiveForIn;

    @JsonProperty("is_active_for_out")
    private boolean isActiveForOut;

    @JsonProperty("country")
    private CountryResponseDTO country;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public boolean getIsActiveForIn() {
        return isActiveForIn;
    }

    public void setIsActiveForIn(boolean activeForIn) {
        isActiveForIn = activeForIn;
    }

    public boolean getIsActiveForOut() {
        return isActiveForOut;
    }

    public void setIsActiveForOut(boolean activeForOut) {
        isActiveForOut = activeForOut;
    }

    public CountryResponseDTO getCountry() {
        return country;
    }

    public void setCountry(CountryResponseDTO country) {
        this.country = country;
    }
}

File: MyLanguageWeb/src/main/resources/static/js/classes/dto/entity/lang/response/lang_response_dto.js
import {
    CountryResponseDTO
} from "../../country/response/country_response_dto.js";

export class LangResponseDTO {
    #id;
    #title;
    #code;
    #isActiveForIn;
    #isActiveForOut;
    #country;

    constructor(json) {
        if (json) {
            this.#id = json["id"];
            this.#title = json["title"];
            this.#code = json["code"];
            this.#isActiveForIn = json["is_active_for_in"];
            this.#isActiveForOut = json["is_active_for_out"];

            let country = json["country"];
            if (country) {
                this.#country = new CountryResponseDTO(country);
            }
        }
    }

    getId() {
        return this.#id;
    }

    getTitle() {
        return this.#title;
    }

    getCode() {
        return this.#code;
    }

    getIsActiveForIn() {
        return this.#isActiveForIn;
    }

    getIsActiveForOut() {
        return this.#isActiveForOut;
    }

    getCountry() {
        return this.#country;
    }
}